src/devtool/solid/App.tsx

```diff
@@ -61,11 +61,34 @@ function App() {
 		}
 	};
 
-	onMount(() => {
+	onMount(async () => {
 		window.handleRunPause = handleRunPause;
 		window.handleStep = handleStep;
 		window.handleReset = handleReset;
 
+		// Wait for WebUI bindings to be ready, then load initial bytecode
+		const loadInitialBytecode = async () => {
+			// Check if WebUI functions are available
+			if (
+				typeof window.load_bytecode === "function" &&
+				typeof window.reset_evm === "function"
+			) {
+				try {
+					await loadBytecode(bytecode());
+					const initialState = await resetEvm();
+					setState(initialState);
+				} catch (err) {
+					setError(`Failed to load initial bytecode: ${err}`);
+				}
+			} else {
+				// Retry after a short delay
+				setTimeout(loadInitialBytecode, 100);
+			}
+		};
+
+		// Start the initial loading process
+		setTimeout(loadInitialBytecode, 100);
```

Comment:
hacky should be a better way to just have webui emit an event or something

Reply:
Ooops forgot to link to event     [event_connected](https://github.com/evmts/guillotine/blob/9f240a5e7d5708317d0dd54cf5e8f07287abe56e/src/devtool/webui/types.zig#L85)